pkg/config/config_template.yaml

```diff
@@ -413,6 +413,126 @@ api_key:
 #
 # secret_backend_timeout: 5
 
+## @param snmp_listener - custom object - optional
+## Creates and schedules a listener to automatically discover your SNMP devices.
+## Discovered devices can then be monitored with the SNMP integration by using
+## the auto_conf.yaml file provided by default.
+#
+# snmp_listener:
+
+  ## @param workers - integer - optional - default: 2
+  ## The number of concurrent tasks used to discover SNMP devices. Increasing this value
+  ## discovers devices faster but at the cost of increased resource consumption.
+  #
+  # workers: 2
+
+  ## @param discovery_interval - integer - optional - default: 3600
+  ## How often to discover new SNMP devices, in seconds. Decreasing this value
+  ## discovers devices faster but at the cost of increased resource consumption.
+  #
+  # discovery_interval: 3600
+
+  ## @param allowed_failures - integer - optional - default: 3
+  ## The number of failed requests to a given SNMP device before removing it from the list of monitored
+  ## devices.
+  ## If a device shuts down, the Agent stops monitoring it after `discovery_interval * allowed_failures` seconds.
+  #
+  # allowed_failures: 3
+
+  ## @param configs - list - required
+  ## The actual configurations used to discover SNMP devices in various subnets.
+  #
+  # configs: 
+    ## @param network - string - required
+    ## The subnet in CIDR format to scan for SNMP devices.
+    ## All unignored IP addresses in the CIDR range are scanned.
+    ## For optimal discovery time, be sure to use the smallest network mask
+    ## possible as is appropriate for your network topology.
+    ## Ex: 10.0.0.0/24
+    #
+    # - network: <NETWORK>
```

Comment:
I think from customer perspective, the current way might be confusing overall.

What do you think about this way: https://github.com/DataDog/datadog-agent/blob/671ff2443cd817f4c1764228c4f4bb942a4444d1/pkg/config/config_template.yaml#L1060-L1102

Reply:
Not sure I understand

Reply:
Something like this https://github.com/DataDog/datadog-agent/blob/efe1d36b8073dff3f1104b5c588d629e42d7b149/pkg/config/config_template.yaml#L937-L973

with description description and example separation to make it easier to user to uncomment and use it.

Reply:
@AlexandreYang Not sure I follow either. If I take this entire block and hit "Uncomment" I get a working configuration - values are uncommented and `##` comments stay as comments, and the `-` list marker works too.

Reply:
Yes, agree, uncommenting the current way should work too.

My suggestion was about providing examples in a more condensed way to make it easier for user to uncomment and use like this (possibly in addition to the current detailed config):

```
  #  - network: <NETWORK>
  #    port: 161
  #    version: 2
  #  - network: <NETWORK>
  #    port: 161
  #    version: 3
  #    user: <USER>
  #    authentication_key:
```

That said, the current way is ok. This is just a nit/niceToHave.